remix-lib/src/execution/txRunner.js

```diff
@@ -146,24 +145,20 @@ class TxRunner {
   }
 
   runBlockInVm (tx, block, callback) {
-    executionContext.vm().runBlock({ block: block, generate: true, skipBlockValidation: true, skipBalance: false }, function (err, results) {
-      err = err ? err.message : err
-      if (err) {
-        return callback(err)
-      }
+    executionContext.vm().runBlock({ block: block, generate: true, skipBlockValidation: true, skipBalance: false }).then(function (results) {
       let result = results.results[0]
-
       if (result) {
-        result.status = '0x' + result.vm.exception.toString(16)
+        result.status = '0x' + (result.execResult.exceptionError ? 0 : 1)
       }
-
       executionContext.addBlock(block)
       executionContext.trackTx('0x' + tx.hash().toString('hex'), block)
-
-      callback(err, {
+      callback(null, {
         result: result,
         transactionHash: ethJSUtil.bufferToHex(Buffer.from(tx.hash()))
       })
+    }).catch(function (err) {
+      err = err ? err.message : err
```

Comment:
I don't really understand why you check if there is an error message. If there is no err then is shouldn't catch shouldn't it ?
Else if `err` is undefined in the catch, then I guess we should create our own error instead of forwarding a callback with undefined.